lib/elixir/lib/string.ex

```diff
@@ -618,6 +618,10 @@ defmodule String do
   @spec replace_leading(t, t, t) :: t
   def replace_leading(string, match, replacement)
       when is_binary(string) and is_binary(match) and is_binary(replacement) do
+    if match == "" do
```

Comment:
It seems like using a guard (or pattern matching in a function) is the usual way of dealing with invalid input.

```elixir
def replace_leading(_string, "", _replacement) do
  raise ArgumentError, "cannot use an empty string as the match to replace"
end

def replace_leading(string, match, replacement)
    when is_binary(string) and is_binary(match) and is_binary(replacement) do
  # implementation
end
```

looks to me the most idiomatic. I would suspect guards to introduce less overhead (although that's marginally important in that case).